test/values.test.js

```diff
@@ -150,6 +150,86 @@ describe('Values', () => {
     expect(json.parseJSON(test_query_json)).toEqual(test_query)
   })
 
+  test('parse versioned lambda agnostic of object order', () => {
+    var simpleQuery = new Query({
+      lambda: 'x',
+      expr: { var: 'x' },
+      api_version: '3',
+    })
+
+    var complexQuery = new Query({
+      lambda: '_',
+      expr: { match: { index: 'idx' }, terms: ['str', 10] },
+      api_version: '3',
+    })
+
+    // simpleQuery assertions
+    var x = '{"@query":{"api_version":"3","lambda":"x","expr":{"var":"x"}}}'
+    expect(json.parseJSON(x)).toEqual(simpleQuery)
+
+    var y = '{"@query":{"lambda":"x","api_version":"3","expr":{"var":"x"}}}'
+    expect(json.parseJSON(y)).toEqual(simpleQuery)
+
+    var z = '{"@query":{"expr":{"var":"x"},"lambda":"x","api_version":"3"}}'
+    expect(json.parseJSON(z)).toEqual(simpleQuery)
+
+    // complexQuery assertions
+    var a =
+      '{"@query":{"lambda":"_","expr":{"match":{"index":"idx"},"terms":["str",10]},"api_version":"3"}}'
+    expect(json.parseJSON(a)).toEqual(complexQuery)
+
+    var b =
+      '{"@query":{"expr":{"match":{"index":"idx"},"terms":["str",10]},"api_version":"3","lambda":"_"}}'
+    expect(json.parseJSON(b)).toEqual(complexQuery)
+
+    var c =
+      '{"@query":{"api_version":"3","expr":{"match":{"index":"idx"},"terms":["str",10]},"lambda":"_"}}'
+    expect(json.parseJSON(c)).toEqual(complexQuery)
+  })
+
+  test('parse functions regardless of order', () => {
```

Comment:
this test is not necessary because the driver doesn't parse the expression, you should focus on the pretty print tests to ensure the queries values render the correct code

Reply:
@marrony I was going to ask you about that 😅 Should I just remove this whole `test` block?

Reply:
fell free to remove